src/components/Widgets/WidgetEvents.tsx

```diff
@@ -85,79 +80,47 @@ export function WidgetEvents() {
-              [TrackingEventParameter.TxLink]: update.process.txLink || '',
-              [TrackingEventParameter.Type]: update.process.type,
-              [TrackingEventParameter.GasCostUSD]:
-                update.route.gasCostUSD || '',
-              [TrackingEventParameter.ErrorCode]:
-                update.process.error?.code || '',
-              [TrackingEventParameter.ErrorMessage]:
-                update.process.error?.message || '',
-              nonInteraction: true,
-            },
+            value: parseFloat(update.route.fromAmountUSD),
+            data,
           });
         }
       }
     };
     const onRouteExecutionCompleted = async (route: Route) => {
-      if (!!route.id) {
-        trackEvent({
+      if (route.id) {
+        const data = handleTransactionDetails(route, {
+          [TrackingEventParameter.Action]: 'execution_completed',
+          [TrackingEventParameter.TransactionStatus]: 'COMPLETED',
+        });
+        trackTransaction({
           category: TrackingCategory.WidgetEvent,
           action: TrackingAction.OnRouteExecutionCompleted,
           label: 'execution_success',
           value: parseFloat(route.fromAmountUSD),
-          data: {
-            [TrackingEventParameter.RouteId]: route.id,
-            [TrackingEventParameter.FromChainId]: route.fromChainId,
-            [TrackingEventParameter.FromAmountUSD]: route.fromAmountUSD,
-            [TrackingEventParameter.FromAmount]: route.fromAmount,
-            [TrackingEventParameter.FromToken]: route.fromToken.address,
-            [TrackingEventParameter.ToChainId]: route.toChainId,
-            [TrackingEventParameter.ToAmountUSD]: route.toAmountUSD,
-            [TrackingEventParameter.ToAmount]: route.toAmount,
-            [TrackingEventParameter.ToAmountMin]: route.toAmountMin,
-            [TrackingEventParameter.ToToken]: route.toToken.address,
-          },
+          data,
           enableAddressable: true,
           isConversion: true,
         });
       }
     };
     const onRouteExecutionFailed = async (update: RouteExecutionUpdate) => {
-      trackEvent({
+      const data = handleTransactionDetails(update.route, {
+        [TrackingEventParameter.Action]: 'execution_failed',
+        [TrackingEventParameter.TransactionStatus]: 'FAILED',
+      });
+      trackTransaction({
         category: TrackingCategory.WidgetEvent,
         action: TrackingAction.OnRouteExecutionFailed,
         label: 'execution_error',
-        data: {
-          [TrackingEventParameter.RouteId]: update.route.id,
-          [TrackingEventParameter.TxHash]: update.process.txHash || '',
-          [TrackingEventParameter.Status]: update.process.status,
-          [TrackingEventParameter.Message]: update.process.message || '',
-          [TrackingEventParameter.ErrorMessage]:
-            update.process.error?.message || '',
-          [TrackingEventParameter.ErrorCode]: update.process.error?.code || '',
-        },
+        value: -1,
```

Comment:
just a question but is -1 expected?

Reply:
Thanks! I removed the "value" attribute, since we have all necessary information available elsewhere in the props.